scripts/serial_link.py

```diff
@@ -334,6 +360,36 @@ def format_log_entry(t0, item):
   delta = int((time.time() - t0)*1000)
   return (delta, timestamp, item)
 
+def format_log_json_entry(t0, item):
+  """
+  Generates a Python object with logging metadata.
+
+  Parameters
+  ----------
+  t0 : float
+    Reference time (seconds)
+  item : object
+    Python object to serialize
+
+  Returns
+  ----------
+  {'delta': delta, 'timestamp': timestamp, 'data': data} : dict
+    delta = msec reference timestamp (int)
+    timestamp = current timestamp (int - UTC epoch)
+    data = Python object to JSON serialize
+  """
+  timestamp = calendar.timegm(time.gmtime())
+  delta = int((time.time() - t0)*1000)
+  data = {
```

Comment:
Agree - I had an iteration done that way... but its more like returning a dict that will be used to JSON serialize - like `to_json_dict` or something - and then doing the actual serialization in the formatter (and adding the extra elements to the envelope).